src/main/java/com/example/carsharing/service/impl/StripePaymentServiceImpl.java

```diff
@@ -0,0 +1,99 @@
+package com.example.carsharing.service.impl;
+
+import com.example.carsharing.dto.payment.CreatePaymentRequestDto;
+import com.example.carsharing.dto.payment.PaymentDto;
+import com.example.carsharing.dto.rental.RentalWithDetailedCarInfoDto;
+import com.example.carsharing.exception.EntityNotFoundException;
+import com.example.carsharing.exception.PaymentException;
+import com.example.carsharing.exception.RentalIsAlreadyPaid;
+import com.example.carsharing.mapper.PaymentMapper;
+import com.example.carsharing.model.Payment;
+import com.example.carsharing.repository.PaymentRepository;
+import com.example.carsharing.service.NotificationService;
+import com.example.carsharing.service.PaymentService;
+import com.example.carsharing.service.RentalService;
+import com.example.carsharing.util.StripeUtil;
+import com.stripe.exception.StripeException;
+import com.stripe.model.checkout.Session;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.List;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class StripePaymentServiceImpl implements PaymentService {
+    private final PaymentRepository paymentRepository;
+    private final PaymentMapper paymentMapper;
+    private final RentalService rentalService;
+    private final StripeUtil stripeUtil;
+    private final NotificationService notificationService;
+
+    @Override
+    public List<PaymentDto> findByUser(Long userId) {
+        return paymentRepository.findAll().stream()
+                .filter(payment -> payment.getRental().getUser().getId().equals(userId))
+                .map(paymentMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public PaymentDto save(CreatePaymentRequestDto requestDto) {
+        Long rentalId = requestDto.rentalId();
+        RentalWithDetailedCarInfoDto rental = rentalService.findById(rentalId);
+        Optional<Payment> byRentalId = paymentRepository.findByRentalId(rentalId);
+        if (byRentalId.isPresent()) {
+            Payment payment = byRentalId.get();
+            if (payment.getStatus().name().equals("PAID")) {
+                throw new RentalIsAlreadyPaid("Rental is already paid!");
+            } else {
```

Comment:
```suggestion
            }
```
no else after throwing is needed